user/advanced-configuration/uefi-troubleshooting.md

```diff
@@ -25,77 +25,6 @@ If you've installed successfully in legacy mode but had to change some kernel pa
-04. Go to `tty2` (Ctrl-Alt-F2).
-05. Enable `/mapbs /noexitboot` on just installed system. This step differs between Qubes releases:
-   
-    **For Qubes 3.1:**
-
-06. Execute `mount | grep boot/efi` and note device name (first column). It should be something like `/dev/sda1`.
-07. Execute `efibootmgr -v`, search for `Qubes` entry and note its number (it should be something like `Boot0001` - `0001` is an entry number).
-08. Replace existing `Qubes` entry with modified one. Replace `XXXX` with entry number from previous step, `/dev/sda` with your disk name and `-p 1` with `/boot/efi` partition number):
-
-        efibootmgr -b XXXX -B
-        efibootmgr -v -c -u -L Qubes -l /EFI/qubes/xen.efi -d /dev/sda -p 1 "placeholder /mapbs /noexitboot"
-
-09. Compare new entry with the old one (printed in step 6) - it should only differ in additional options at the end, and look probably something like this:
-
-        Boot0001* Qubes HD(1,GPT,partition-guid-here,0x800,0x64000)/File(\EFI\qubes\xen.efi)p.l.a.c.e.h.o.l.d.e.r. ./.m.a.p.b.s. ./.n.o.e.x.i.t.b.o.o.t.
-
-    If instead it looks like:
-
-        Boot0001* Qubes HD(1,0,00000000...0,0x0,0x0)/File(\EFI\qubes\xen.efi)p.l.a.c.e.h.o.l.d.e.r. ./.m.a.p.b.s. ./.n.o.e.x.i.t.b.o.o.t.
-
-    then try passing `/dev/sda1` or `/dev/nvme0n1p1` or whatever your EFI partition is instead of `/dev/sda` and `-p 1`.
-
-10. Now you can reboot the system by issuing `reboot` command.
-
-    **For Qubes 3.2 or later:**
-
-11. Edit `/mnt/sysimage/boot/efi/EFI/qubes/xen.cfg` (you can use `vi` editor) and add to every kernel section:
-            
-        mapbs=1
-        noexitboot=1
-
-    **Note:** You must add these parameters on two separate new lines (one
-    parameter on each line) at the end of each section that includes a kernel
-    line (i.e., all sections except the first one, since it doesn't have a
-    kernel line).
-
-12. Now you can reboot the system by issuing `reboot` command.
-
-
-System crash/restart when booting installer
--------------------------------------------
-
-Some Dell systems and probably others have [another bug in UEFI firmware](http://markmail.org/message/amw5336otwhdxi76). And there is another workaround for it:
-
-
-1. In GRUB menu<sup id="a1-2">[1](#f1)</sup> press `e`.
-2. At the end of `chainloader` line add `-- efi=attr=uc`.
-3. Perform installation normally, but don't reboot the system at the end yet.
-4. Go to `tty2` (Ctrl-Alt-F2).
-5. Execute:
-
-        sed -i -e 's/^options=.*/\0 efi=attr=uc/' /mnt/sysimage/boot/efi/qubes/xen.cfg
-        
-   or if you're installing 3.2 execute:
-   
-        sed -i -e 's/^options=.*/\0 efi=attr=uc/' /mnt/sysimage/boot/efi/EFI/qubes/xen.cfg
-
-6. Now you can reboot the system by issuing `reboot` command.
-
-* * *
-<b name="f1">1</b> If you use rEFInd, you can see 3 options regarding the USB installer. Choose "Fallback Boot Loader" to enter the GRUB menu. [↩](#a1-1) [↩](#a1-2)
-
-
```

Comment:
Those two entries do apply to R4.0 too. The only difference is how to add Xen parameters (there is no grub on efi) - the way described above those two points (by editing config file).

Reply:
Aren't `mapbs=1`, `noexitboot=1`, and `efi=attr=uc` all enabled by default in the R4.0 installer?

Reply:
They are on the installer, but not on installed system. If your system require them, you can still boot the installer as kind of rescue image (or do that just after installation, if you know that beforehand) and add them to the target /boot/efi

Reply:
I see. I'd like to move them after the installer boot troubleshooting sections then, and make them generic. I'll start over with a different pull.